rasa/core/channels/twilio_voice.py

```diff
@@ -0,0 +1,342 @@
+            sender_id = request.form.get("From")
+            text = request.form.get("SpeechResult")
+            input_channel = self.name()
+            call_status = request.form.get("CallStatus")
+
+            collector = TwilioVoiceCollectingOutputChannel()
+
+            # Provide an initial greeting to answer the user's call.
+            if (text is None) and (call_status == "ringing"):
+                text = self.initial_prompt
+
+            # determine the response.
+            if text is not None:
+                await on_new_message(
+                    UserMessage(
+                        text, collector, sender_id, input_channel=input_channel,
+                    )
+                )
+
+                twilio_response = self.build_twilio_voice_response(collector.messages)
+            # If the user doesn't respond resend the last message.
+            else:
+                # Get last user utterance from tracker.
+                tracker = request.app.agent.tracker_store.retrieve(sender_id)
+                last_response = next(
+                    (e for e in reversed(tracker.events) if isinstance(e, BotUttered)),
+                    None,
+                )
+
+                # If no previous utterance found use the reprompt_fallback phrase.
+                if last_response is None:
+                    last_response = self.reprompt_fallback_phrase
+                else:
+                    last_response = last_response.text
+
+                twilio_response = self.build_twilio_voice_response(
+                    [{"text": last_response}]
+                )
+            return response.text(str(twilio_response), content_type="text/xml")
+
+        return twilio_voice_webhook
+
+    def build_twilio_voice_response(
+        self, messages: List[Dict[Text, Any]]
+    ) -> VoiceResponse:
+        """Builds the Twilio Voice Response object."""
+        voice_response = VoiceResponse()
+        gather = Gather(
+            input="speech",
+            action=f"/webhooks/{self.name()}/webhook",
+            actionOnEmptyResult=True,
+            speechTimeout=self.speech_timeout,
+            speechModel=self.speech_model,
+            enhanced=self.enhanced,
+        )
+
+        # Add pauses between messages.
+        # Add a listener to the last message to listen for user response.
+        for i, message in enumerate(messages):
+            if i + 1 == len(messages):
+                gather.say(message["text"], voice=self.assistant_voice)
+                voice_response.append(gather)
+            else:
+                voice_response.say(message["text"], voice=self.assistant_voice)
+                voice_response.pause(length=1)
+
+        return voice_response
+
+
+class TwilioVoiceCollectingOutputChannel(CollectingOutputChannel):
+    """Output channel that collects send messages in a list.
+
+    (doesn't send them anywhere, just collects them).
+    """
+
+    EMOJI = re.compile(
```

Comment:
we have sth similar [here](https://github.com/RasaHQ/rasa/blob/20f0d70d6cbad9e3a3a4db23a7b8df5d08bc173d/rasa/nlu/tokenizers/whitespace_tokenizer.py#L41). How about re-using it?